src/uwtools/config.py

```diff
@@ -170,15 +142,13 @@ def dereference(
         self, ref_dict: Optional[dict] = None, full_dict: Optional[dict] = None
     ) -> None:
         """
-        This method will be used as a method by which any Config object can cycle through its
-        key/value pairs recursively, replacing Jinja2 templates as necessary.
-        """
+        Recursively replace Jinja2 templates in a Config object.
 
-        if ref_dict is None:
-            ref_dict = self.data
-
-        if full_dict is None:
-            full_dict = self.data
+        :param ref_dict: ???
+        :param full_dict: ???
+        """
+        ref_dict = self.data if ref_dict is None else ref_dict
+        full_dict = self.data if full_dict is None else full_dict
```

Comment:
Hm, good point! It looks like a bit of a workaround to not cause issues if only one is specified, but I'm not sure thats necessarily how we want to do it now that you mention it

Reply:
When `dereference` is first called, it would be called without arguments on `self.data`, which might be a YAML dict read in with Jinja2 templates in the value. We want to `dereference` the whole dictionary, so set it to `ref_dict` for that purpose. 

Now, we want to be able to fill in the Jinaj2 templates with any of the values in the `self.data` Jinja2 templates with respect to the full dictionary -- `self.data`.

On recursion, `dereference` gets passed the nested dictionaries in the full dictionary, but also the full dictionary itself so that Jinja2 templates can reference the structure from the top level. In those recursive calls, we'd be passing dicts explicitly.

We could likely get away with replacing `full_dict` with references to `self.data`.

Reply:
I have updated the parameter descriptions and added a line explaining that the arguments are generally meant to be supplied when the method recursively calls itself. I think this is good for now.

FWIW, a pattern I've seen that handles this kind of situation is to have a main, public, potentially no-argument method/function that hands off to a private method/function that takes care of the recursion. For example, we might have `dereference() -> None` that then calls `_dereference(ref_dict: dict, full_dict: dict) -> None`, which also calls itself recursively. Then no arguments are optional and there can't be any confusion at the API level about how to call the method.